projects/plugins/jetpack/modules/custom-post-types/portfolios.php

```diff
@@ -555,39 +667,44 @@ function allow_portfolio_rest_api_type( $post_types ) {
 	 * Our [portfolio] shortcode.
 	 * Prints Portfolio data styled to look good on *any* theme.
 	 *
+	 * @param array $atts Shortcode attributes.
+	 *
 	 * @return portfolio_shortcode_html
 	 */
-	static function portfolio_shortcode( $atts ) {
-		// Default attributes
-		$atts = shortcode_atts( array(
-			'display_types'   => true,
-			'display_tags'    => true,
-			'display_content' => true,
-			'display_author'  => false,
-			'show_filter'     => false,
-			'include_type'    => false,
-			'include_tag'     => false,
-			'columns'         => 2,
-			'showposts'       => -1,
-			'order'           => 'asc',
-			'orderby'         => 'date',
-		), $atts, 'portfolio' );
-
-		// A little sanitization
-		if ( $atts['display_types'] && 'true' != $atts['display_types'] ) {
-			$atts['display_types'] = false;
-		}
-
-		if ( $atts['display_tags'] && 'true' != $atts['display_tags'] ) {
-			$atts['display_tags'] = false;
-		}
+	public static function portfolio_shortcode( $atts ) {
+		// Default attributes.
+		$atts = shortcode_atts(
+			array(
+				'display_types'   => true,
+				'display_tags'    => true,
+				'display_content' => true, // Can be false, true, or full.
+				'display_author'  => false,
+				'show_filter'     => false,
+				'include_type'    => false,
+				'include_tag'     => false,
+				'columns'         => 2,
+				'showposts'       => -1,
+				'order'           => 'asc',
+				'orderby'         => 'date',
+			),
+			$atts,
+			'portfolio'
+		);
 
-		if ( $atts['display_author'] && 'true' != $atts['display_author'] ) {
-			$atts['display_author'] = false;
-		}
+		/*
+		 * A little sanitization for our shortcode attributes aiming to use booleans.
+		 * Attributes can be booleans (from the default values) or strings.
+		 */
+		foreach ( $atts as $attribute_name => $attribute_value ) {
+			if ( preg_match( '#^display_|include_|show_#i', $attribute_name ) ) {
+				// display_content is a special case.
+				if ( 'display_content' === $attribute_name && 'full' === $attribute_value ) {
+					$atts['display_content'] = 'full';
+					continue;
+				}
 
-		if ( $atts['display_content'] && 'true' != $atts['display_content'] && 'full' != $atts['display_content'] ) {
-			$atts['display_content'] = false;
+				$atts[ $attribute_name ] = self::sanitize_boolean_attribute( $attribute_value );
+			}
```

Comment:
This is the part that I refactored, as I mentioned in the PR description. It allows for avoiding a good number of phpcs:ignore later in the file, and makes things clearer in my opinion, but if there is a better way or if this seems overkill, let me know.